x/ccv/provider/keeper/distribution.go

```diff
@@ -57,32 +70,201 @@ func (k Keeper) GetAllConsumerRewardDenoms(ctx sdk.Context) (consumerRewardDenom
+		powerFraction := math.LegacyNewDec(vote.Validator.Power).QuoTruncate(math.LegacyNewDec(totalPower))
+		tokensFraction := tokens.MulDecTruncate(powerFraction)
+
+		k.distributionKeeper.AllocateTokensToValidator(
+			ctx,
+			k.stakingKeeper.ValidatorByConsAddr(ctx, consAddr),
+			tokensFraction,
+		)
+		totalReward = totalReward.Add(tokensFraction...)
+	}
+
+	return totalReward
+}
+
+// TransferConsumerRewardsToDistributionModule transfers the collected rewards of the given consumer chain
+// from the consumer rewards pool module account to a the distribution module
+func (k Keeper) TransferConsumerRewardsToDistributionModule(
+	ctx sdk.Context,
+	chainID string,
+) (sdk.Coins, error) {
+	// Get coins of the consumer rewards allocation
+	allocation := k.GetConsumerRewardsAllocation(ctx, chainID)
+
+	if allocation.Rewards.IsZero() {
+		return sdk.Coins{}, nil
+	}
+
+	// Truncate coin rewards
+	rewardsToSend, _ := allocation.Rewards.TruncateDecimal()
+
+	// NOTE the consumer rewards allocation isn't a module account, however its coins
+	// are held in the consumer reward pool module account. Thus the consumer
+	// rewards allocation must be reduced separately from the SendCoinsFromModuleToAccount call.
+
+	// Update consumer rewards allocation with the remaining decimal coins
+	allocation.Rewards = allocation.Rewards.Sub(sdk.NewDecCoinsFromCoins(rewardsToSend...))
+
+	// Send coins to distribution module account
+	err := k.bankKeeper.SendCoinsFromModuleToModule(ctx, types.ConsumerRewardsPool, distrtypes.ModuleName, rewardsToSend)
+	if err != nil {
+		return sdk.Coins{}, err
+	}
+
+	k.SetConsumerRewardsAllocation(ctx, chainID, allocation)
+	return rewardsToSend, nil
+}
+
+// consumer reward pools getter and setter
+
+// GetConsumerRewardsAllocation returns the consumer rewards allocation for the given chain ID
+func (k Keeper) GetConsumerRewardsAllocation(ctx sdk.Context, chainID string) (pool types.ConsumerRewardsAllocation) {
+	store := ctx.KVStore(k.storeKey)
+	b := store.Get(types.ConsumerRewardsAllocationKey(chainID))
+	k.cdc.MustUnmarshal(b, &pool)
+	return
+}
+
+// SetConsumerRewardsAllocation sets the consumer rewards allocation for the given chain ID
+func (k Keeper) SetConsumerRewardsAllocation(ctx sdk.Context, chainID string, pool types.ConsumerRewardsAllocation) {
+	store := ctx.KVStore(k.storeKey)
+	b := k.cdc.MustMarshal(&pool)
+	store.Set(types.ConsumerRewardsAllocationKey(chainID), b)
+}
+
+// GetConsumerRewardsPool returns the balance
+// of the consumer rewards pool module account
+func (k Keeper) GetConsumerRewardsPool(ctx sdk.Context) sdk.Coins {
+	return k.bankKeeper.GetAllBalances(
+		ctx,
+		k.accountKeeper.GetModuleAccount(ctx, types.ConsumerRewardsPool).GetAddress(),
+	)
+}
+
+// ComputeConsumerTotalVotingPower returns the total voting power for a given consumer chain
+// by summing its opted-in validators votes
+func (k Keeper) ComputeConsumerTotalVotingPower(ctx sdk.Context, chainID string, votes []abci.VoteInfo) int64 {
```

Comment:
This method doesn't need the votes but rather the consumer valset. For now leave it like this, but it needs to be updated once we have the consumer valsets in store.